fairlearn/metrics/_disaggregated_result.py

```diff
@@ -39,14 +39,22 @@ def extract_unique_classes(
 
 
 def apply_to_dataframe(
-    data: pd.DataFrame, metric_functions: Dict[str, AnnotatedMetricFunction]
+    data: pd.DataFrame,
+    metric_functions: Dict[str, AnnotatedMetricFunction],
+    include_groups: bool = False,
 ) -> pd.Series:
     """Apply metric functions to a DataFrame.
 
     The incoming DataFrame may have been sliced via `groupby()`.
     This function applies each annotated function in turn to the
     supplied DataFrame.
+
+    The include_groups argument is weird. It appears that pandas
+    introduced it as an argument in v2.2, and immediately deprecated
+    it (dependent on when this is being read, may need to adjust):
+    https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.apply.html
     """
+    assert not include_groups, "Unexpected change to include_groups"
```

Comment:
can we avoid assert in our code? I don't understand the motivation behind this line.

Reply:
I found that the `include_groups` argument was being included in the invocation of this function. But per `pandas` docs, it should only ever be False. So if it isn't, something really weird is happening; I just wanted to check on this. I can tweak the message

Reply:
but it's not used in this function at all

Reply:
Correct. It's needed to work with pandas<2.2 (which I don't think we're dropping yet?). I've removed the `assert` as requested.